src/erlcloud_aws.erl

```diff
@@ -363,6 +370,39 @@ config_env() ->
         _ -> {error, environment_config_unavailable}
     end.
 
+config_with_region(Config) ->
+    case {os:getenv("AWS_REGION"), application:get_env(erlcloud, aws_region)} of
```

Comment:
I'm not opposed to it, as long as it's backwards-compatible and well-documented.